LinhNT/Linux/Understand-how-to-use-a-Linux-shell.md

```diff
@@ -0,0 +1,185 @@
++ **./script.sh** : đọc **!#** để xác định chương trình để chạy script.sh và cần được **cấp quyền**. 
+
+#### 2. Biến trong shell
+**Đặt tên**
++ Tên biến chỉ gồm chữ cái: a-z, A-Z, 0-9, "_"
++ Tên biến không được bắt đầu bằng số
+```
+Name="LinhNT"
+```
+**Sử dụng**
+```
+Name="LinhNT"
+echo $Name
+```
+
+**Biến read-only**
+
+![](https://github.com/linhnt31/internship-2020/blob/linhnt-baocao-t1/LinhNT/Linux/Images-how-to-use-shell/readonly-var.PNG)
+
+**Xóa giá trị của biến**
+
+![](https://github.com/linhnt31/internship-2020/blob/linhnt-baocao-t1/LinhNT/Linux/Images-how-to-use-shell/unset-var.PNG)
+
+**Các loại biến**
++ Biến hệ thống : tạo ra, quản lí bởi hệ thống và được viết hoa.
+![](https://github.com/linhnt31/internship-2020/blob/linhnt-baocao-t1/LinhNT/Linux/Images-how-to-use-shell/types-of-var.PNG)
+
++ Biến do người dùng tự định nghĩa: tạo và định nghĩa bởi người dùng.
+**Mảng**
+Có 2 cách khai báo mảng: 
+*Input* 
+```
+#!/bin/bash
+NAME[0]="Zara"
+NAME[1]="Qadir"
+NAME[2]="Mahnaz"
+NAME[3]="Ayan"
+NAME[4]="Daisy"
+echo "First Index: ${NAME[0]}"
+echo "Second Index: ${NAME[1]}"
+echo "First Method: ${NAME[*]}"
+echo "Second Method: ${NAME[@]}"
+```
+
+![](https://github.com/linhnt31/internship-2020/blob/linhnt-baocao-t1/LinhNT/Linux/Images-how-to-use-shell/array.PNG)
+
+hoặc
+```
+#!/bin/bash
+NAME=(a b c d e)
+echo "First Index: ${NAME[0]}"
+echo "Second Index: ${NAME[1]}"
+echo "First Method: ${NAME[*]}"
+echo "Second Method: ${NAME[@]}"
+```
+
+![](https://github.com/linhnt31/internship-2020/blob/linhnt-baocao-t1/LinhNT/Linux/Images-how-to-use-shell/array-declaration.PNG)
+
++ Để sử dụng một giá trị trong màng: **${array_name[index]}**
++ Để in ra toàn bộ mảng: **{array_name[*]}** hoặc **{array_name[@]}**
+
+**Các phép toán số học**
+*Input*
+```
+!#/bin/bash
+
+a=10
+b=3
+cong=$(($a + $b))
+tru=$(($a - $b))
+nhan=$(($a * $b))
+chia=$(($a / $b))
+layDu=$(($a % $b))
+
+ganGiaTri=$b
+
+echo $cong
+echo $tru
+echo $nhan
+echo $chia
+echo $layDu
+echo $ganGiaTri
+```
+
+*Output*
+![](https://github.com/linhnt31/internship-2020/blob/linhnt-baocao-t1/LinhNT/Linux/Images-how-to-use-shell/math.PNG)
+
+**Cấu trúc rẽ nhánh**
+*Input* : conditions.sh
+```
+read -p "Nhap mot so > 20: " n
+if [ $n -gt 20 ];
+then
+        echo "$n lon hon 20"
+elif [ $n == 20 ];
+then 
+        echo "$n bang 20"
+else 
+        echo "ban nhap so nho hon 20"
+fi
+```
+
+OR
+
+```
+read -p "Nhap mot so > 20: " n
+if test $n -gt 20
+then
+        echo "$n lon hon 20"
+else 
+        echo "ban nhap so nho hon 20"
+fi
+```
+*Output*
+
+![](https://github.com/linhnt31/internship-2020/blob/linhnt-baocao-t1/LinhNT/Linux/Images-how-to-use-shell/conditions-results-1.PNG)
+
+**Cấu trúc lặp**
+```
+#!/bin/bash
+a=0
+while [ $a -lt 10 ]
+do
+        echo $a
+        if [ $a -eq 5 ];
+        then 
+                break
+        fi
+        a=$((a+1))
+        sleep 1
+done
+```
+
+OR
```

Comment:
Ý em **OR** ở đây là 2 script giống nhau hay thế nào nhỉ?

Reply:
OR ở đây em muốn thực hiện là mỗi đoạn code hay bài toán thì sẽ có nhiều cách để thực hiện ạ.  Như ở đây kiểm tra điều kiện ([ $a -gt x], test $a -gt x,..), vòng lặp thì có thể sử dụng từ khóa khác nhau (while, until,..),. Em đã chỉnh sửa 2 đoạn scripts lại đồng bộ rồi ạ